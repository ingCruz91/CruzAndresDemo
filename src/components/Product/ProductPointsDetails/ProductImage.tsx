import React from 'react';
import {StyleSheet, View} from 'react-native';
import FastImage from 'react-native-fast-image';

interface ProductImage {
  productImage: string;
}

const ProductImage = ({productImage}: ProductImage) => {
  return (
    <View style={styles.imageContainer}>
      <FastImage
        testID="productImage"
        style={styles.image}
        source={{
          uri: productImage,
          priority: FastImage.priority.normal,
        }}
        resizeMode={FastImage.resizeMode.contain}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  imageContainer: {
    marginVertical: 10,
    shadowOpacity: 0.5,
    shadowRadius: 5,
    shadowOffset: {
      width: 0,
      height: 10,
    },
    elevation: 5,
    backgroundColor: 'white',
    borderWidth: 0.1,
    flex: 2.8,
    borderRadius: 10,
    marginHorizontal: 15,
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    borderRadius: 5,
    width: 200,
    height: 200,
    marginRight: 10,
  },
});

export default ProductImage;
