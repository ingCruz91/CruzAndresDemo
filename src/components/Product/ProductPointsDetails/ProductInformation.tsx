import React from 'react';
import {StyleSheet, View} from 'react-native';
import CustomLabel from '../../custom/CustomLabel';
import {PRODUCT_LABELS} from '../../../Utils/constants';

interface ProductInformation {
  productCreatedAt: string;
  productPoints: number;
}

const ProductInformation = ({
  productCreatedAt,
  productPoints,
}: ProductInformation) => {
  return (
    <View style={styles.detailsContainer}>
      <CustomLabel.GrayBold size={14} style={styles.productDetails}>
        {PRODUCT_LABELS.DETALLES_DEL_PRODUCTO}
      </CustomLabel.GrayBold>
      <CustomLabel.BlackBold size={14} style={styles.saleDate}>
        {productCreatedAt}
      </CustomLabel.BlackBold>
      <CustomLabel.GrayBold size={14} style={styles.earnedPoints}>
        {PRODUCT_LABELS.CON_ESTA_COMPRA_ACUMULASTE}
      </CustomLabel.GrayBold>
      <CustomLabel.BlackBold size={22} style={styles.earnedPoints}>
        {productPoints} {PRODUCT_LABELS.PUNTOS}
      </CustomLabel.BlackBold>
    </View>
  );
};

const styles = StyleSheet.create({
  detailsContainer: {
    flex: 1.7,
    justifyContent: 'flex-start',
  },
  productDetails: {
    marginLeft: 15,
    marginTop: 17,
  },
  saleDate: {
    marginLeft: 15,
    marginVertical: 19,
  },
  earnedPoints: {
    marginLeft: 15,
    marginBottom: 25,
  },
});

export default ProductInformation;
