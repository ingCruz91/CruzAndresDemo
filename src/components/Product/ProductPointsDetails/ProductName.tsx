import React from 'react';
import {StyleSheet, View} from 'react-native';
import CustomLabel from '../../custom/CustomLabel';

interface ProductName {
  productName: string;
}

const ProductName = ({productName}: ProductName) => {
  return (
    <View style={styles.productNameContainer}>
      <CustomLabel.BlackBold size={22} style={styles.productName}>
        {productName}
      </CustomLabel.BlackBold>
    </View>
  );
};

const styles = StyleSheet.create({
  productNameContainer: {
    marginBottom: 5,
    backgroundColor: '#CFD6FF',
    flex: 0.8,
    justifyContent: 'flex-end',
  },
  productName: {
    marginLeft: 10,
    marginBottom: 15,
  },
});

export default ProductName;
