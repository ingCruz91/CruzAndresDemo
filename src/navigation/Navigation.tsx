import React from 'react';
import {createStackNavigator} from '@react-navigation/stack';
import {NavigationContainer} from '@react-navigation/native';
import ProductPoints from '../screens/ProductPoints';
import ProductPointsDetails from '../screens/ProductPointsDetails';

export type RootStackParamList = {
  ProductPoints: undefined;
  ProductPointsDetails: {
    productName: string;
    productCreatedAt: string;
    productPoints: number;
    productImage: string;
  };
};

const Stack = createStackNavigator<RootStackParamList>();
const screenOptions = {
  headerShown: false,
  gestureEnabled: false,
  swipeEnabled: false,
};
const Navigation = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator
        initialRouteName="ProductPoints"
        screenOptions={screenOptions}>
        <Stack.Screen name="ProductPoints" component={ProductPoints} />
        <Stack.Screen
          name="ProductPointsDetails"
          component={ProductPointsDetails}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default Navigation;
