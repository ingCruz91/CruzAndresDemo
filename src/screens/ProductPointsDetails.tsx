import * as React from 'react';
import {View, StyleSheet, SafeAreaView} from 'react-native';
import {RouteProp, useNavigation} from '@react-navigation/native';
import {RootStackParamList} from '../navigation/Navigation';
import {StackNavigationProp} from '@react-navigation/stack';
import ProductButton from '../components/Product/ProductButton';
import ProductName from '../components/Product/ProductPointsDetails/ProductName';
import ProductImage from '../components/Product/ProductPointsDetails/ProductImage';
import ProductInformation from '../components/Product/ProductPointsDetails/ProductInformation';
import {PRODUCT_LABELS} from '../Utils/constants';

export type ProductPointsDetailsProps = {
  route: RouteProp<RootStackParamList, 'ProductPointsDetails'>;
};

export function ProductPointsDetails({route}: ProductPointsDetailsProps) {
  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();

  const {productName, productCreatedAt, productPoints, productImage} =
    route.params;

  return (
    <SafeAreaView style={styles.appContainer}>
      <ProductName productName={productName} />
      <ProductImage productImage={productImage} />
      <ProductInformation
        productCreatedAt={productCreatedAt}
        productPoints={productPoints}
      />
      <View style={styles.buttonsContainer}>
        <ProductButton
          style={styles.buttonStyle}
          onPress={() => navigation.pop()}>
          {PRODUCT_LABELS.ACEPTAR}
        </ProductButton>
      </View>
    </SafeAreaView>
  );
}

export default ProductPointsDetails;

const styles = StyleSheet.create({
  appContainer: {
    backgroundColor: '#EAEAEA',
    flex: 1,
  },
  buttonsContainer: {
    flex: 0.8,
    alignItems: 'stretch',
    justifyContent: 'center',
  },
  buttonStyle: {
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 10,
    height: 45,
    marginHorizontal: 20,
  },
});
