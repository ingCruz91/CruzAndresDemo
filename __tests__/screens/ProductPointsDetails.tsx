import React from 'react';
import {render} from '@testing-library/react-native';
import ProductPointsDetails from '../../src/screens/ProductPointsDetails';
import {NavigationContainer, RouteProp} from '@react-navigation/native';
import {RootStackParamList} from '../../src/navigation/Navigation';

const route: RouteProp<RootStackParamList, 'ProductPointsDetails'> = {
  params: {
    productName: 'string',
    productCreatedAt: 'string',
    productPoints: 0,
    productImage: 'string',
  },
  key: '123',
  name: 'ProductPointsDetails',
  path: '../src/screens/ProductPointsDetails',
};

describe('ProductPointsDetails', () => {
  it('renders ProductPointsDetails correctly', () => {
    const {toJSON} = render(
      <NavigationContainer>
        <ProductPointsDetails route={route} />
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
