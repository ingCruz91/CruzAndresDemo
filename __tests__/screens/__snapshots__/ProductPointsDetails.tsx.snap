// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductPointsDetails renders ProductPointsDetails correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#EAEAEA",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#CFD6FF",
        "flex": 0.8,
        "justifyContent": "flex-end",
        "marginBottom": 5,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#020202",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 22,
          },
          {
            "marginBottom": 15,
            "marginLeft": 10,
          },
        ]
      }
    >
      string
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "borderWidth": 0.1,
        "elevation": 5,
        "flex": 2.8,
        "justifyContent": "center",
        "marginHorizontal": 15,
        "marginVertical": 10,
        "shadowOffset": {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.5,
        "shadowRadius": 5,
      }
    }
  >
    <View
      style={
        [
          {
            "overflow": "hidden",
          },
          {
            "borderRadius": 5,
            "height": 200,
            "marginRight": 10,
            "width": 200,
          },
        ]
      }
    >
      <FastImageView
        defaultSource={null}
        resizeMode="contain"
        source={
          {
            "priority": "normal",
            "uri": "string",
          }
        }
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="productImage"
      />
    </View>
  </View>
  <View
    style={
      {
        "flex": 1.7,
        "justifyContent": "flex-start",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#9B9898",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 14,
          },
          {
            "marginLeft": 15,
            "marginTop": 17,
          },
        ]
      }
    >
      Detalles del Producto
    </Text>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#020202",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 14,
          },
          {
            "marginLeft": 15,
            "marginVertical": 19,
          },
        ]
      }
    >
      string
    </Text>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#9B9898",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 14,
          },
          {
            "marginBottom": 25,
            "marginLeft": 15,
          },
        ]
      }
    >
      Con esta compra acumulaste:
    </Text>
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#020202",
            "fontFamily": "AvenirBlack",
            "fontWeight": "bold",
          },
          {
            "fontFamily": undefined,
            "fontSize": 22,
          },
          {
            "marginBottom": 25,
            "marginLeft": 15,
          },
        ]
      }
    >
      0
       
       puntos
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "stretch",
        "flex": 0.8,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "stretch",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#334FFA",
            "borderRadius": 10,
            "height": 45,
            "justifyContent": "center",
            "marginHorizontal": 20,
            "opacity": 1,
          }
        }
      >
        <Text>
          <Text
            style={
              [
                {
                  "color": "white",
                  "fontWeight": "bold",
                },
                undefined,
              ]
            }
          >
            Aceptar
          </Text>
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
